jib-core/src/main/java/com/google/cloud/tools/jib/image/json/ImageToJsonTranslator.java

```diff
@@ -61,11 +62,22 @@
     if (exposedPorts == null) {
       return null;
     }
+
+    return listToMap(
+        exposedPorts
+            .stream()
+            .map(port -> port.getPort() + "/" + port.getProtocol())
```

Comment:
I think an alternative way to do this is to have a list to map like:

```
static <E> Map<String, Map<?, ?>> listToMap(@Nullable List<E> list, Function<E, String> listElementToString)
```

where this method will directory turn the list of E into a map of string to `{}`, where each element E converts to String using the `listElementToString` function.

This can avoid what is going on here where the list is turned into a stream, then mapped, then collected into a list, then iterated over with foreach to place into an `ImmutableSortedMap`. ([`ImmutableSortedMap.toImmutableSortedMap`](https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/ImmutableSortedMap.html#toImmutableSortedMap-java.util.Comparator-java.util.function.Function-java.util.function.Function-) may be useful here.)

Reply:
Thanks for the suggestion. I'm not familiarized with Immutable collections specific methods, that's nice to know.

Reply:
I honestly wasn't aware of `ImmutableSortedMap.toImmutableSortedMap` till I looked it up haha - learning new things everyday.